Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean

```diff
@@ -0,0 +1,124 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Homology.ShortComplex.Exact
+import Mathlib.CategoryTheory.Abelian.Basic
+import Mathlib.CategoryTheory.Shift.ShiftSequence
+import Mathlib.CategoryTheory.Triangulated.Functor
+import Mathlib.CategoryTheory.Triangulated.Subcategory
+
+/-! # Homological functors
+
+In this file, given a functor `F : C ⥤ A` from a pretriangulated category to
+an abelian category, we define the type class `F.IsHomological`, which is the property
+that `F` sends distinguished triangles in `C` to exact sequences in `A`.
+
+If `F` is a homological functor, we defined the strictly full triangulated subcategory
+`F.homologicalKernel`.
+
+## TODO
+
+* The long exact sequence in homology attached to an homological functor.
+
+## References
+* [Jean-Louis Verdier, *Des catégories dérivées des catégories abéliennes*][verdier1996]
+
+-/
+
+namespace CategoryTheory
+
+open Category Limits Pretriangulated ZeroObject Preadditive
+
+variable {C D A : Type*} [Category C] [HasZeroObject C] [HasShift C ℤ] [Preadditive C]
+  [∀ (n : ℤ), (CategoryTheory.shiftFunctor C n).Additive] [Pretriangulated C]
+  [Category D] [HasZeroObject D] [HasShift D ℤ] [Preadditive D]
+  [∀ (n : ℤ), (CategoryTheory.shiftFunctor D n).Additive] [Pretriangulated D]
+  [Category A] [Abelian A]
+
+namespace Functor
+
+variable (F : C ⥤ A)
+
+/-- A functor from a pretriangulated category to an abelian category is an homological functor
+if it sends distinguished triangles to exact sequences. -/
+class IsHomological extends F.PreservesZeroMorphisms : Prop where
```

Comment:
I think this is usually called "cohomological" in English?
cf: https://ncatlab.org/nlab/show/cohomological+functor

Reply:
I have added a terminological note in the docstring of the file.